sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1621,6 +1623,17 @@ def handle_cmis_state_machine(self, lport, info, is_fast_reboot):
             log_exception_traceback()
             self.update_port_transceiver_status_table_sw_cmis_state(lport, CMIS_STATE_FAILED)
 
+    def do_task(self, is_fast_reboot):
+        for lport, info in self.port_dict.items():
+            if self.task_stopping_event.is_set():
+                break
+
+            self.handle_cmis_state_machine(lport, info, is_fast_reboot)
+
+        for event in self.deleted_ports.values():
+            self.port_dict.pop(event.port_name)
```

Comment:
Would it make the flow/logic simpler by just directly deleting `self.port_dict[lport]` entry at the PORT_DEL event of CONFIG_DB in `on_port_update_event()`?

Reply:
Done.

I explained the background of this change [here](https://github.com/sonic-net/sonic-platform-daemons/pull/500#discussion_r1762172401)